pkg/reconciler/subscription/subscription.go

```
@@ -373,32 +378,36 @@ func (r *Reconciler) ensureFinalizer(sub *v1alpha1.Subscription) error {
 }
 
 // resolveResult resolves the Spec.Result object.
-func (r *Reconciler) resolveResult(ctx context.Context, namespace string, replyStrategy *v1alpha1.ReplyStrategy, track eventingduck.Track) (string, error) {
+func (r *Reconciler) resolveResult(ctx context.Context, namespace string, replyStrategy *v1alpha1.ReplyStrategy, trackAddressable eventingduck.Track) (string, error) {
 	if isNilOrEmptyReply(replyStrategy) {
 		return "", nil
 	}
 
-	// Tell tracker to reconcile this Subscription whenever the Channel changes.
-	if err := track(*replyStrategy.Channel); err != nil {
+	// Tell tracker to reconcile this Subscription whenever the reply Channel changes. Note that it does not need to be
+	// a Channel, it needs to be an Addressable. As Channels are Addressable, we are fine.
+	if err := trackAddressable(*replyStrategy.Channel); err != nil {
 		logging.FromContext(ctx).Error("Unable to track changes to spec.reply.channel", zap.Error(err))
 		return "", err
 	}
 
-	obj, err := eventingduck.ObjectReference(ctx, r.DynamicClientSet, namespace, replyStrategy.Channel)
+	lister, err := r.addressableTracker.ListerFor(*replyStrategy.Channel)
 	if err != nil {
-		logging.FromContext(ctx).Warn("Failed to fetch ReplyStrategy Channel",
-			zap.Error(err),
-			zap.Any("replyStrategy", replyStrategy))
-		return "", err
+		logging.FromContext(ctx).Error(fmt.Sprintf("Error getting lister for ObjecRef: %s/%s", namespace, replyStrategy.Channel.Name), zap.Error(err))
```

We still need to return an error here.

---

Done, thanks for the catch